src/dfinstancewindows.cpp

```diff
@@ -89,9 +89,9 @@ QString DFInstanceWindows::read_string(const uint &addr) {
     Q_ASSERT_X(len < (1 << 16), "read_string",
                "String must be of sane length!");
 
-    char buf[len];
-    read_raw(buffer_addr, len, buf);
-    return QTextCodec::codecForName("IBM437")->toUnicode(buf, len);
+    std::unique_ptr<char[]> buf = std::make_unique<char[]>(len);

```

Comment:
As discussed in #320, we want to avoid doing heap allocation in a hot path. The fix I have pending avoids heap allocation and uses VLAs if supported.